nemo/collections/nlp/models/machine_translation/mt_enc_dec_model.py

```diff
@@ -280,6 +286,29 @@ def _setup_dataloader_from_config(self, cfg: DictConfig):
                 raise ValueError("src must be equal to target for cached dataset")
             dataset = pickle.load(open(cfg.src_file_name, 'rb'))
             dataset.reverse_lang_direction = cfg.get("reverse_lang_direction", False)
+        elif cfg.get("load_from_tarred_dataset", False):
+            logging.info('Loading from tarred dataset %s' % (cfg.src_file_name))
+            if cfg.src_file_name != cfg.tgt_file_name:
+                raise ValueError("src must be equal to target for tarred dataset")
+            if not cfg.get("metadata_path", ""):
+                raise FileNotFoundError("Could not find metadata path in config")
+            dataset = TarredTranslationDataset(
+                text_tar_filepaths=cfg.src_file_name,
+                metadata_path=cfg.metadata_path,
+                encoder_tokenizer=self.encoder_tokenizer,
+                decoder_tokenizer=self.decoder_tokenizer,
+                shuffle_n=cfg.get("shuffle_n", 100),
+                shard_strategy=cfg.get("shard_strategy", "scatter"),
+                global_rank=self.global_rank,
+                world_size=self.world_size
+            )
+            return torch.utils.data.DataLoader(
+                dataset=dataset,
+                batch_size=1,
+                num_workers=cfg.get("num_workers", 2),
```

Comment:
You need to set the collate function for typechecking to work (when you add types that is)

Reply:
Not sure what I should be adding types to here?

Reply:
Let's skip it for now, it can be added later if needed.